### DISCUSSION TITLE
Adding trackers via Tramission Web?

### DISCUSSION THREAD

**Discussion by lewri** [transmission/transmission#6727] [Q&A] [answered]
Edit: There is actually an issue open (https://github.com/transmission/transmission/issues/5234) that aligns closely enough to this query to achieve the same goal.

I have always been a user of transmission with web interface over the years on single board devices. I do greatly enjoy it!

I am not the most component user of search terms on the issues page but last year I saw that Transmission 4 was released with a remade WebUI. I am excited to see it. However, from what I can see I can't actually find a list of function or feature changes (if any).
I believe the readme file for Web is still targeting older versions: https://github.com/transmission/transmission/blob/main/docs/Web-Interface.md

So my question is, **has the new version added a method to add trackers to a torrent via the web interface?** As I currently don't know this, and my current usage is on a not so rollback-able NAS, I haven't made the leap yet from version 3.00-20. Currently I use a modified version of the web interface with https://github.com/54mv39/web as this is still compatible on 3.00-20, I don't believe it would be compatible with version 4. An image just to show how the modified interface acts:
![image](https://github.com/transmission/transmission/assets/20030128/f2000585-da3e-4b0c-ac97-3d142dbf36f8)

**Comment by tearfur** [comment, marked as answer]
> has the new version added a method to add trackers to a torrent via the web interface?

Not as far as I am aware. Please create a new feature request issue for it!

**Reply by lewri** [reply]
Thanks I've raised this now on #6756

**Reply by lewri** [reply]
@tearfur https://github.com/transmission/transmission/issues/5234 has been suggested as a duplicate of the ticket I raised (I agree), if you wanted to take a look.

**Comment by Pentaphon** [comment]
You can easily do this with another web-UI like https://github.com/openscopeproject/TrguiNG which I enjoy using and has excellent backwards compatibility with older versions of Transmission for people who can't update. I also find this UI much easier and more straightforward to use than the old and new official WebUIs.

**Comment by killemov** [comment]
@Pentaphon That UI MAY be used to easily add trackers to a torrent, but it adds another layer of software server-side.

@lewri You could have been using  [Shift](https://github.com/killemov/Shift) for over a decade now. It is a pure web-UI replacement for all concurrent versions of Transmission. Just download/clone, change one environment variable and off you go!
```
Environment=TRANSMISSION_WEB_HOME=/my/path/to/the/unzipped/dir
```

**Reply by Pentaphon** [reply]
>That UI MAY be used to easily add trackers to a torrent, but it adds another layer of software server-side.

Not completely true. It's also a pure Web-UI replacement like Shift. Pay attention to the project page and you can see this is the case.

>You can use this program in 2 ways: as a native Windows/Linux/Mac app and as a web gui served by transmission itself by setting $TRANSMISSION_WEB_HOME environment variable to point to TrguiNG web assets. 

See?

**Reply by killemov** [reply]
> Not completely true.

From https://github.com/openscopeproject/TrguiNG :
Backend for the app is written in [rust](https://www.rust-lang.org/).

So what functionality is present in what form? What does it need a backend for?

**Reply by Pentaphon** [reply]
>So what functionality is present in what form? What does it need a backend for?

I don't know, but I just proved to you that it is also a pure WebUI like yours. That's all that matters here. Hopefully yours or the one I use might be useful to the OP.

**Reply by killemov** [reply]
>  also a pure WebUI

So by definition it is not.

**Reply by lewri** [reply]
Although I appreciate both these insights I think it detracts from the point (more specifically as an enhancement request of #6756) here in that I was asking specifically about the built-in WebUI, rather than a facelifted alternative access method. No disrespect intended. #6756 then in that regard is about asking it as a feature of Transmission's in house WebUI.

Edit: to note -- Shift actually isn't so much to my liking as the in house Transmission WebUI, but thanks for bringing it to my attention.
TrguiNG is likely compatible with Synology DSM but not straightforwardly so.

**Reply by Pentaphon** [reply]
>I think it detracts from the point (more specifically as an enhancement request of https://github.com/transmission/transmission/issues/6756) here in that I was asking specifically about the built-in WebUI, rather than a facelifted alternative access method. No disrespect intended. https://github.com/transmission/transmission/issues/6756 then in that regard is about asking it as a feature of Transmission's in house WebUI.

I understand but since this is a discussion rather than that issue I thought it would be helpful to provide existing alternatives than waiting for that feature to be added to the official WebUI. I've thumbed up your issue so that it gets more attention and I hope others do the same as I do agree that the official WebUI should include that basic functionality.
